shapely/constructive.py

```diff
@@ -888,6 +890,90 @@ def minimum_bounding_circle(geometry, **kwargs):
 
     See also
     --------
-    minimum_bounding_radius
+    minimum_bounding_radius, maximum_inscribed_circle
     """
     return lib.minimum_bounding_circle(geometry, **kwargs)
+
+
+@requires_geos("3.9.0")
+@multithreading_enabled
+def maximum_inscribed_circle(geometry, tolerance=0, **kwargs):
+    """Finds the largest circle that is fully contained within the input geometry.
+
+    Constructs the "maximum inscribed circle" (MIC) for a polygonal geometry,
+    up to a specified tolerance. The MIC is determined by a point in the
+    interior of the area which has the farthest distance from the area
+    boundary, along with a boundary point at that distance. In the context of
+    geography the center of the MIC is known as the "pole of inaccessibility".
+    A cartographic use case is to determine a suitable point to place a map
+    label within a polygon.
+    The radius length of the MIC is a  measure of how "narrow" a polygon is.
+    It is the distance at which the negative buffer becomes empty.
+
+    The class supports polygons with holes and multipolygons.
+
+    Returns a two-point linestring, with the first point at the center of the
+    inscribed circle and the second on the boundary of the inscribed circle.
+
+    Parameters
+    ----------
+    geometry : Geometry or array_like
+    tolerance : float or array_like
+        Stop the algorithm when the search area is smaller than this tolerance
```

Comment:
In PostGIS, this is not exposed as an argument to the user, but always calculated based on the size of the geometry (https://github.com/postgis/postgis/blob/6f335e8117bcbc43daa9e0536c0703e3926f8acd/postgis/lwgeom_geos.c#L395-L398):

```
		if (!gserialized_get_gbox_p(geom, &gbox))
			PG_RETURN_NULL();

		width = gbox.xmax - gbox.xmin;
		height = gbox.ymax - gbox.ymin;
		size = width > height ? width : height;
		tolerance = size / 1000.0;
```

So even if we still expose the keyword (if we think that is useful), we could use similar logic for the default value if it was not specified by the user.

Reply:
I think that is a very sensible thing to do.